### PR TITLE
Replaced `for` loop with enhanced `for` loop.

### PR DESCRIPTION
**Pull request by DiegoKrupitza**
In few cases the simple `for` loop can be replaced with a enhanced `for` loop to improve readability and make the code cleaner.

### CODE REVIEW

File: src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java [hunk 1]
```diff
@@ -637,8 +637,7 @@ private Object writeReplace() {
       size = input.length; // implicit nullcheck of input
 
       elements = (E[]) new Object[EXPAND_FACTOR * input.length];
-      for (int i = 0; i < input.length; i++) {
-        E e = input[i];
+      for (E e : input) {
```
Comment (scordio) [inline src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java:640]: Please revert

File: src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java [hunk 2]
```diff
@@ -1173,8 +1172,8 @@ private void readObject(ObjectInputStream ois) throws IOException, ClassNotFound
   private void writeObject(ObjectOutputStream oos) throws IOException {
     oos.defaultWriteObject();
     oos.writeInt(array.length);
-    for (int i = 0; i < array.length; i++) {
-      oos.writeObject(array[i]);
+    for (Object o : array) {
+      oos.writeObject(o);
```
Comment (scordio) [inline src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java:1176]: Please revert

**Review by scordio — COMMENTED** [review]
Thanks @DiegoKrupitza, please do not apply changes to `ImmutableCollections` as it has been sourced from the JDK 11.

File: src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java (hunk 1 above)
Comment (DiegoKrupitza) [reply src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java:640]: Oh... I totally overlooked it. Will revert it asap!

Comment (DiegoKrupitza) [reply src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java]: Reverted

File: src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java (hunk 2 above)
Comment (DiegoKrupitza) [reply src/test/java/org/assertj/core/test/jdk11/ImmutableCollections.java]: Reverted

**Pull request merged**